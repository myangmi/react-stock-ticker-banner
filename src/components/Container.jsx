import React from 'react';
import PropTypes, { string } from 'prop-types';
import View from './View.jsx';

import './Container.scss';

const mockTickers = ['AMD', 'MSFT', 'AMZN', 'INTC', 'TSM', 'AAPL', 'FB', 'NFLX'];

const propTypes = {
  tickers: PropTypes.arrayOf(string),
  apiKey: PropTypes.string,
};

const Container = ({ tickers, apiKey }) => (
  <div className='Container'>
    <View tickers={mockTickers} apiKey={apiKey} />
  </div>
);

Container.propTypes = propTypes;
export default Container;
