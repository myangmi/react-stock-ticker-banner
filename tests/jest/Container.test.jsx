import React from 'react';
import { shallow } from 'enzyme';

import Container from '../../src/components/Container.jsx';
import mockTickers from '../mocks/mockTickers';

describe('Container', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Container tickers={mockTickers} apiKey={'mockAPIKey'}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
