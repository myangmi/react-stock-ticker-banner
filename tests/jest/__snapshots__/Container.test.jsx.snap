// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render properly 1`] = `
<div
  className="Container"
>
  <View
    apiKey="mockAPIKey"
    tickers={
      Array [
        "AMD",
        "INTC",
        "TSM",
        "AAPL",
      ]
    }
  />
</div>
`;
